File: unix_utilities/wcat.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAXLINE 1024
#define MAXWORD 25

int main(int argc, char **argv)
{
    // probably should print an error message
    // but directions want it just to exit returning 0
    if (argc < 2) exit(0);
    
    FILE *fp;
    char buf[MAXLINE];

    while (--argc > 0 && (*argv)) {
        fp = fopen(*++argv, "r");
        if (fp == NULL) {
            printf("wcat: cannot open file\n");
            exit(1);
        }
        while (fgets(buf, MAXLINE, fp)) {
            sleep(1);
            printf("%s", buf);
            memset(buf, 0, MAXLINE);
        }
    }
    return 0;
}

File: unix_utilities/wcat.py
import sys

def main():

    if (len(sys.argv) < 2):
        for line in sys.stdin:
            print(line, end='')

    i = 1
    while (i < len(sys.argv)):
        try:
            fp = open(sys.argv[i], 'r')
            for line in fp:
                print(line, end='')
        except:
            print('wcat: cannot open file')
            sys.exit(1)
        i+=1

if __name__ == '__main__':
    main()
